doc/continuous-integration.md

```diff
@@ -0,0 +1,33 @@
+# Continuous integration
+
+## Main
+
+The [govuk_smart_answers CI instance](https://ci-new.alphagov.co.uk/job/govuk_smart_answers/) is triggered by new commits on the `master` branch of the [GitHub repo](https://github.com/alphagov/smart-answers). It runs the [`jenkins.sh`](https://github.com/alphagov/smart-answers/blob/master/jenkins.sh) script with the `RUN_REGRESSION_TESTS` not set. This script does the following:
+
+* Checks out the lastest [govuk-content-schemas](https://github.com/alphagov/govuk-content-schemas) which are used in a few of the tests
+* Installs the bundled gems
+* Runs govuk-lint-ruby (see [Rubocop docs](doc/rubocop.md))
+* Runs the `test` rake task with `TEST_COVERAGE` enabled
+* Pre-compiles assets
+
+If any of these steps fails, the build fails-fast.
+
+## Branches
+
+The [govuk_smartanswers_branches CI instance](https://ci-new.alphagov.co.uk/job/govuk_smartanswers_branches/) is triggered by new commits on other branches of the [GitHub repo](https://github.com/alphagov/smart-answers). It runs the [`jenkins_branches.sh`](https://github.com/alphagov/smart-answers/blob/master/jenkins_branches.sh) script with the `RUN_REGRESSION_TESTS` not set.
+
+This script does the same as the [main CI instance](#main), but first it tries to merge `master` into the relevant branch in order to work out whether it will merge without conflicts. This merge is always aborted, but if conflicts are detected the script fails fast at this point.
```

Comment:
I'm not sure this is quite right.

1. It's the jenkins.sh script that merges master into the branch, which just happens to be a no-op if the branch is master.

2. I don't think the merge is always aborted, and I think this is why we've ended up with Jenkins merge commits in our release tags.